lib/buckets/buckets.js

```diff
@@ -265,9 +272,54 @@ function createObjectMetadata(req, type, cb) {
     }
 }
 
-function notFoundHandler(req, res, next) {
-    if (req.not_found_error) {
-        next(req.not_found_error);
+/*
+ * Handles the 200->304 response code translation of some HTTP requests.  All
+ * other precondition cases are handled by buckets-mdapi and will result in a
+ * PreconditionFailedError error already being passed through the server's
+ * response pipeline and thus not reaching this point.
+ *
+ * In particular, only GET and HEAD requests are subject to this translation,
+ * and only when the "If-None-Match" header is present and matched upon, or in
+ * its absence "If-Modified-Since" is after the object's "Last-Modified".
+ */
+function conditionalHandler(req, res, next) {
+    assert.object(req.conditions, 'req.conditions');
+
+    var if_modified_since, if_none_match;
+    var code;
+
+    if (!isConditional(req) ||
+        (req.method !== 'HEAD' && req.method !== 'GET')) {
+        next();
+        return;
+    }
+
+    if ((if_none_match = req.conditions['if-none-match'])) {
+        var object_etag = res.header('Etag');
+
+        if_none_match.forEach(function (client_etag) {
+            if (client_etag === '*' || client_etag === object_etag) {
+                code = 304;
+                return;
+            }
+        });
+    }
+
+    if ((if_modified_since = req.conditions['if-modified-since'])) {
+        /*
+         * XXX What do we do here if for some reason buckets-mdapi has passed us
```

Comment:
That's a good question. In normal circumstances this should not ever happen based on the postgres column type used and the serialization in `buckets-mdapi` so if this fails maybe the right thing is to return a 500 to indicate something has really gone wrong?

Reply:
Actually it looks like this assumption is already made when we initially set the headers (see [here](https://github.com/joyent/manta-buckets-api/blob/0310aa0c4b4f1bc480f7b4c8b5a1e6d78cec2f8b/lib/buckets/objects/metadata/update.js#L72) or [here](https://github.com/joyent/manta-buckets-api/blob/0310aa0c4b4f1bc480f7b4c8b5a1e6d78cec2f8b/lib/buckets/objects/head.js#L69) for examples). The reason this method is doing another parse of the date is because when setting the "Last-Modified" header in the above cases it's cast as a string, so we must re-parse it.

I've removed the `XXX` comment for now, but I'd still like to handle this case while setting the headers. There were a few minor inconsistencies that I started ironing out as part of MANTA-4556 (or as part of the followup I mentioned in the ticket), so I'd be happy to add this scenario to that ticket if that works for you.